### ISSUE TITLE
Correct Way to Implement Frame Stacking

### ISSUE THREAD

**Issue by lukepolson** [tensorflow/agents#334]
I currently have an environment where the observation shape is (1,40,40,1) (first element is the batch dimension). I want to frame stack the last few frames so that my observations are (1,40,40,3) but HistoryWrapper makes them (1,3,40,40,1). What is the correct way to frame stack such that the observations are (1,40,40,3)? I've considered using a keras preprocessing_layer that swaps axis=1 and axis=4 making the observations (1,1,40,40,3) and then removing the first axis, but this feels hackish. I feel like there's a more appropriate way of doing this.

**Comment by lukepolson** [comment]
My hackish method is to use the following preprocessing layer:

`preprocessing_layer = Sequential([`        
`tf.keras.layers.Permute((4,2,3,1)),`
`tf.keras.layers.Lambda(lambda x: x[:,0,:,:,:])`
`])`

This seems to work, but I feel like its not the correct way to use history wrapper

**Comment by kbanoop** [comment]
Does this work for you? https://github.com/tensorflow/agents/blob/213cc66b0adef6a44231769f77f88bdb80759d1c/tf_agents/environments/atari_wrappers.py#L29

**Comment by tfboyd** [comment]
Closing due to being stale.
